src/adapters/third-party-providers-service-fetcher.ts

```diff
@@ -23,15 +23,29 @@ export async function createThirdPartyProvidersServiceFetcherComponent(
   const isThirdPartyProvidersResolverServiceDisabled: boolean =
     (await config.getString('DISABLE_THIRD_PARTY_PROVIDERS_RESOLVER_SERVICE_USAGE')) === 'true'
 
-  return {
-    async get(): Promise<ThirdPartyProvider[]> {
-      if (isThirdPartyProvidersResolverServiceDisabled) {
-        throw new Error(
-          'Third Party Providers resolver service will not be used since DISABLE_THIRD_PARTY_PROVIDERS_RESOLVER_SERVICE_USAGE is set'
-        )
+  async function get(): Promise<ThirdPartyProvider[]> {
+    if (isThirdPartyProvidersResolverServiceDisabled) {
+      throw new Error(
+        'Third Party Providers resolver service will not be used since DISABLE_THIRD_PARTY_PROVIDERS_RESOLVER_SERVICE_USAGE is set'
+      )
+    }
+    const response: ThirdPartyProvidersServiceResponse = await (await fetch.fetch(`${serviceUrl}/providers`)).json()
+
+    if (response.thirdPartyProviders) {
+      for (const thirdParty of response.thirdPartyProviders) {
+        if (thirdParty.metadata.thirdParty?.contracts) {
+          thirdParty.metadata.thirdParty.contracts = thirdParty.metadata.thirdParty.contracts.map((c) => ({
+            network: c.network.toLowerCase(),
+            address: c.address.toLowerCase()
+          }))
+        }
       }
-      const response: ThirdPartyProvidersServiceResponse = await (await fetch.fetch(`${serviceUrl}/providers`)).json()
-      return response.thirdPartyProviders
     }
```

Comment:
WDYT about moving this logic to a util function? There is something similar in `src/adapters/third-party-providers-graph-fetcher.ts`

Reply:
Done.